### PR TITLE
Basket changes review

### PR DESCRIPTION
**Pull request by EvanBarbour3**
## Features
- Actions on view pages have been changed to be included inside views
- Models linked to each other
- Active flag has been changed to a slider on edit page
- Confirmation screens have been made 'pretty'
- Messages moved to master
- Logo has been changed
- 'Gravatar' bug fixed

[Delivers#101042528]

### CODE REVIEW

File: app/Exceptions/RedirectException.php [hunk 1]
```diff
@@ -112,4 +112,17 @@ public function setInformation($information)
         $this->information = $information;
         return $this;
     }
+
+    /**
+     * @author WN
+     * @return array
+     */
+    public function toArray()
+    {
```
Comment (wnowicki) [inline app/Exceptions/RedirectException.php]: You should implement rest fields on this method.

BTW you don't need to add my initials ;)

File: app/Http/Controllers/ApplicationsController.php [hunk 2]
```diff
@@ -313,6 +310,7 @@ private function renderConfirmationScreen($action, $id)
         if (((!$this->isCancellable($application)) && $action == 'cancellation') ||
             ((!$this->isFulfilable($application)) && $action == 'fulfilment')
         ) {
+            Log::error('Application is not allowed to request ' . $action);
```
Comment (wnowicki) [inline app/Http/Controllers/ApplicationsController.php]: Why you removed this?

File: app/Http/Middleware/Messages.php [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+namespace App\Http\Middleware;
+
+class Messages
+{
+    public function handle($request, $next)
+    {
+        view()->share('messages', is_array(session()->get('messages'))?session()->get('messages'):[]);
+        return $next($request);
+    }
+}
```
Comment (wnowicki) [inline app/Http/Middleware/Messages.php]: CS

File: resources/views/applications/fulfilment.blade.php [hunk 4]
```diff
@@ -2,39 +2,70 @@
...
-                    <div style="right: 15px" class="pull-right col-sm-3 col-xs-4">
-                        <a href="{{Request::server('HTTP_REFERER')}}" class="btn btn-info form-control">Cancel</a>
-                    </div>
-                </div>
-            </div>
+            <a href="{{Request::server('HTTP_REFERER')}}" class="btn btn-info">Cancel</a>
         </div>
         {!! Form::close() !!}
+
     {{--@endif--}}
```
Comment (wnowicki) [inline resources/views/applications/fulfilment.blade.php]: ??

File: tests/Controllers/ApplicationsControllerTest.php [hunk 5]
```diff
@@ -41,4 +41,19 @@ public function testMake()
...
+     * Test Show Page
+     *
+     * Check page exists
+     *
+     * @author EB
+     */
+    public function testShowPage()
+    {
+        // Test page gives 200 response
+        $this->visit('applications/show')
```
Comment (wnowicki) [inline tests/Controllers/ApplicationsControllerTest.php]: ???

Comment (EvanBarbour3) [inline tests/Controllers/ApplicationsControllerTest.php]: Not a clue, removed

File: resources/views/applications/fulfilment.blade.php (hunk 4 above)
Comment (EvanBarbour3) [inline resources/views/applications/fulfilment.blade.php]: Removed the extra line, believe it is there due to changing the fulfilment page around and making it less heavy.

File: app/Http/Middleware/Messages.php (hunk 3 above)
Comment (EvanBarbour3) [inline app/Http/Middleware/Messages.php]: Added docblocks and extra line at the end

File: app/Exceptions/RedirectException.php (hunk 1 above)
Comment (EvanBarbour3) [inline app/Exceptions/RedirectException.php]: Fields have been added.

**Pull request merged**